tests/testsuite/features_namespaced.rs

```diff
@@ -600,48 +590,84 @@ fn json_exposed() {
-                      "documentation": null,
-                      "source": null,
-                      "dependencies": "{...}",
-                      "targets": "{...}",
-                      "features": {
-                        "bar": ["dep:bar"]
-                      },
-                      "manifest_path": "[..]foo/Cargo.toml",
-                      "metadata": null,
-                      "publish": null,
-                      "authors": [],
-                      "categories": [],
-                      "default_run": null,
-                      "keywords": [],
-                      "readme": null,
-                      "repository": null,
-                      "rust_version": null,
-                      "edition": "2015",
-                      "links": null
-                    }
-                  ],
-                  "workspace_members": "{...}",
-                  "workspace_default_members": "{...}",
-                  "resolve": null,
-                  "target_directory": "[..]foo/target",
-                  "version": 1,
-                  "workspace_root": "[..]foo",
-                  "metadata": null
-                }
-            "#,
+        .with_stdout_data(
+            str![[r#"
+{
+  "metadata": null,
+  "packages": [
+    {
+      "authors": [],
+      "categories": [],
+      "default_run": null,
+      "dependencies": [
+        {
+          "features": [],
+          "kind": null,
+          "name": "bar",
+          "optional": true,
+          "registry": null,
+          "rename": null,
+          "req": "^1.0",
+          "source": "registry+https://github.com/rust-lang/crates.io-index",
+          "target": null,
+          "uses_default_features": true
+        }
+      ],
+      "description": null,
+      "documentation": null,
+      "edition": "2015",
+      "features": {
+        "bar": [
+          "dep:bar"
+        ]
+      },
+      "homepage": null,
+      "id": "path+[ROOTURL]/foo#0.1.0",
+      "keywords": [],
+      "license": null,
+      "license_file": null,
+      "links": null,
+      "manifest_path": "[ROOT]/foo/Cargo.toml",
+      "metadata": null,
+      "name": "foo",
+      "publish": null,
+      "readme": null,
+      "repository": null,
+      "rust_version": null,
+      "source": null,
+      "targets": [
+        {
+          "crate_types": [
+            "lib"
+          ],
+          "doc": true,
+          "doctest": true,
+          "edition": "2015",
+          "kind": [
+            "lib"
+          ],
+          "name": "foo",
+          "src_path": "[ROOT]/foo/src/lib.rs",
+          "test": true
+        }
+      ],
+      "version": "0.1.0"
+    }
+  ],
+  "resolve": null,
+  "target_directory": "[ROOT]/foo/target",
+  "version": 1,
+  "workspace_default_members": [
+    "path+[ROOTURL]/foo#0.1.0"
+  ],
+  "workspace_members": [
+    "path+[ROOTURL]/foo#0.1.0"
+  ],
+  "workspace_root": "[ROOT]/foo"
+}
+
+"#]]
+            .json(),
```

Comment:
I am currently manually pretty-printing the json output. However, as the same reason described in https://github.com/rust-lang/cargo/pull/14091/files#r1644803562 we might want to avoid atm https://github.com/rust-lang/cargo/issues/14039#issuecomment-2160827084.

Reply:
For a single JSON object, feel like it shall be fine.